cli/src/main/java/hudson/util/QuotedStringTokenizer.java

```diff
@@ -282,8 +282,8 @@ else if (c == '\\')
     /* ------------------------------------------------------------ */
     @Override
     public String nextToken()
-        throws NoSuchElementException
-    {
+    
+   {
```

Comment:
Hi @MarkEWaite NoSuchElementException does not need to be propagated because it inherits from the RuntimeException class. Not propagating exceptions that are propagated automatically simplifies reading the code. This is the purpose of this Sonar rule.

Reply:
We're still going to analyse this formatting problem because it's not the expected behaviour.